docs/anti-patterns.mdx

```diff
@@ -66,52 +184,159 @@ pub contract Array {
+The admin resource can then be `link()`ed to a private path in an admin's account storage during the contract's initializer.
 
-### Public Capability Fields Are A Security Hole
+### Example
 
-This is a specific case of "Public Functions And Fields Should Be Avoided", above.
+```cadence
+// Pseudo-code
+
+// BAD
+pub contract Currency {
+    pub resource Admin {
+        pub fun mintTokens()
+    }
 
-The values of public fields can be copied. Capabilities are value types. Anyone who receives a Capability can use it. So if they can copy it from a public field they can call the functions that it exposes.
+    // Anyone in the network can call this function
+    // And use the Admin resource to mint tokens
+    pub fun createAdmin(): @Admin {
+        return <-create Admin()
+    }
+}
 
-This almost certainly is not what you want if a capability has been stored in this way. For public access to a capability, place it in an accounts public area.
+// This contract makes the admin creation private and in the initializer
+// so that only the one who controls the account can mint tokens
+// GOOD
+pub contract Currency {
+    pub resource Admin {
+        pub fun mintTokens()
+
+        // Only an admin can create new Admins
+        pub fun createAdmin(): @Admin {
+            return <-create Admin()
+        }
+    }
 
-### Users Might Rebind Restricted Public Capability Paths
+    init() {
+        // Create a single admin resource
+        let firstAdmin <- create Admin()
 
-A public capability on a user account that is a variant of a standard resource (such as a Vault) that has additional logic implemented to restrict or alter its behaviour can be replaced by that user at the same path with another resource that does not have those alterations.
+        // Store it in private account storage in `init` so only the admin can use it
+        self.account.save(<-firstAdmin, to: /storage/currencyAdmin)
+    }
+}
+```
 
-To fix this, make sure to carefully specify the type of the capability that you are expecting in any code that accesses it.
+## Do Not Modify Smart Contract State or Emit Events in public struct initializers
 
-## Readability
+This is another example of the risks of having publicly accessible parts to your smart contract.
 
-### Descriptive Names Are Preferable
+### Problem
 
-`account` / `accounts` is better than `array` / `element`.
+Data structure definitions in Cadence currently must be declared as public so that they can be used by anyone.
+Structs do not have the same restrictions that resources have on them,
+which means that anyone can create a new instance of a struct without going through any authorization.
 
-`providerAccount` / `tokenRecipientAccount` is better than `acct1` / `acct2`.
+### Solution
 
-Unless of course you are dealing with entities that really are best named that way, for example in utility code.
+Any contract state-modifying operations related to the creation of structs
```

Comment:
Great point, did not realize this could potentially be a problem. We could add a linting pass for that